src/compiler/checker.ts

```diff
@@ -11090,7 +10969,7 @@ module ts {
             }
         }
 
-        function getExportNameSubstitution(symbol: Symbol, location: Node): string {
+        function getExportNameSubstitution(symbol: Symbol, location: Node, getGeneratedNameForNode: (Node: Node) => string): string {

```

Comment:
maybe ```, generateName: NameGenerator```  along with ```type NameGenerator = (node: Node) => string;``` ?

I find it pretty tough to read these sigs with function types in it.  But it's up to you.